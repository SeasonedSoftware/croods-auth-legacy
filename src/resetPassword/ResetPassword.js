import React from 'react'
import New from '../New'

export default props => {
  const { relativePath = '/auth/password', location } = props

  return (
    <New
      {...props}
      method="put"
      name="auth.resetPassword"
      path={relativePath}
      otherProps={{ location }}
    />
  )
}
