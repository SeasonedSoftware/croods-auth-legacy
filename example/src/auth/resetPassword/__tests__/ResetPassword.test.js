import React from 'react'
import renderer from 'react-test-renderer'
import { New } from 'croods'
import { navigate } from '@reach/router'

import ResetPassword from '../ResetPassword'

jest.mock('croods', () => ({
  New: props => <div {...props}>New - {props.render(props)}</div>,
}))
jest.mock('../Form', () => props => <div {...props}>Form</div>)

jest.mock('@reach/router', () => ({
  navigate: jest.fn(),
}))

const tree = renderer.create(<ResetPassword />)

it('renders correctly', () => {
  expect(tree.toJSON()).toMatchSnapshot()
})

describe('renderCreated', () => {
  const newObject = tree.root.findByType(New)
  it('calls navigate', () => {
    newObject.props.renderCreated()
    expect(navigate).toBeCalled()
  })
})
