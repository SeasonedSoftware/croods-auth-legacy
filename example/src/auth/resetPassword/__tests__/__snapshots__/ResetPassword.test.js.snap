// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  method="put"
  name="resetPassword"
  path="/admin_auth/password"
  render={[Function]}
  renderCreated={[Function]}
>
  New - 
  <div>
    Form
  </div>
</div>
`;
